allennlp/models/model.py

```diff
@@ -116,12 +117,37 @@ def forward_on_instance(self, instance: Instance, cuda_device: int) -> Dict[str,
         outputs = self.decode(self.forward(**model_input))
 
         for name, output in list(outputs.items()):
+            # We are predicting on a single instance and we added a batch
+            # dimension, so here we remove it.
             output = output[0]
             if isinstance(output, torch.autograd.Variable):
                 output = output.data.cpu().numpy()
             outputs[name] = output
         return outputs
 
+    def forward_on_batch(self, instances: List[Instance],
```

Comment:
`forward(inputs)` already operates on batches, so this name is potentially confusing.  I think something like `forward_on_instances` would be better, or `forward_on_instance_batch` if you really want to keep the word "batch" in there.

Reply:
Also, do you have strong feelings about putting the first argument on the same line as `self`?  I kind of think it should go on the next line.